### ISSUE TITLE
Could two sha256 calculations happen in parallel?

### ISSUE THREAD

**Issue by jans23** [Nitrokey/nitrokey-fido-u2f-firmware#8]

**Comment by szszszsz** [comment]
The main concern is, why these two identical implementations are separately supplied to the firmware:
- in atecc508a:
[atecc508a.c#L215](https://github.com/Nitrokey/nitrokey-fido-u2f-firmware/blob/master/firmware/src/atecc508a.c#L215)
[atecc508a.c#L229](https://github.com/Nitrokey/nitrokey-fido-u2f-firmware/blob/master/firmware/src/atecc508a.c#L229)
- in u2f_atecc:
[u2f_atecc.c#L110](https://github.com/Nitrokey/nitrokey-fido-u2f-firmware/blob/master/firmware/src/u2f_atecc.c#L110)
[u2f_atecc.c#L124](https://github.com/Nitrokey/nitrokey-fido-u2f-firmware/blob/master/firmware/src/u2f_atecc.c#L124)

When one was removed and calls redirected to other, the device has stopped accepting attestation key during SETUP phase: https://github.com/Nitrokey/nitrokey-fido-u2f-firmware/commit/dc835162518eeef4a24babfb1f1fbe099a332077. It had to be reverted to make it work again: https://github.com/Nitrokey/nitrokey-fido-u2f-firmware/commit/51350a4ccb610bc9492a2718c006f802bda752cc, and the symbols were hidden to compilation unit scope via `static`: https://github.com/Nitrokey/nitrokey-fido-u2f-firmware/commit/e27038dfeb24ee54fd999741a70994bcc4a284f6.

Either local state (given via `SHA_FLAGS` and `SHA_HMAC_KEY`) is not cleaned after one of the operations, or SHA calculations are done concurrently, one during another.

- [ ] Pack state variables into structure and store separate copies in both compilation units. 
- [ ] Make u2f_sha functions work on created structure in object-like manner.
- [ ] Remove duplicated code.
- [ ] Investigate, why the implementation was doubled in the first place.

**Comment by szszszsz** [comment]
There is a guess (CC @adamglozik) that `compute_key_hash` may be causing the issue. 
See:  https://github.com/Nitrokey/nitrokey-fido-u2f-firmware/blob/c4f3d6ad5de87c0a96f931373c23ff2b9052c695/firmware/src/u2f_atecc.c#L173

**Comment by szszszsz** [comment]
Refactoring tasks moved to a separate issue.
Cause lied in a missed hardcoded slot number within `compute_key_hash` function.
Fix merged to `master` already. Closing.
